File: Services/src/main/java/project/services/IServices.java
package project.services;

import project.model.Flight;
import project.model.FlightDTO;
import project.model.User;


import java.util.List;

public interface IServices {

    public Integer login(String email, String password, IObserver client) throws ServiceException;
    void logout(User user, IObserver client) throws ServiceException;
    public List<Flight> getAllFlights();
    //public Iterable<Ticket> getAllTickets();
    //public List<Flight> getFlightBySearch(String searchDestination, LocalDateTime searchDateTime);
    public List<Flight> getFlightBySearchAirport(String searchDestination, String airport) throws  ServiceException;
    //public Flight addFlight(Flight elem);
    //public User addUser(User user);
    public void buyTicket(Integer flightId, String name, String tourists, String address, Integer seats) throws ServiceException;
}

File: Services/IServices.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Model;

namespace Services
{
	public interface IServices
	{

		int login(String email, String password, IObserver client);

		void logout(User user, IObserver client);

		IList<Flight> getAllFlights();

		//IList<Ticket> getAllTickets();

		//IList<Flight> getFlightBySearch(String searchDestination, DateTime searchDateTime);

		IList<Flight> getFlightBySearchAirport(String searchDestination, String airport);

		//void addFlight(Flight elem);

		//void addUser(User user);

		void buyTicket(int flightId, String name, String tourists, String address, int seats);
	}
}
